static/app/views/dashboards/widgets/timeSeriesWidget/releaseBubbles/utils/createReleaseBuckets.tsx

```diff
@@ -56,19 +64,34 @@ export function createReleaseBuckets(
       break;
     }
 
-    // We can determine the releaase's bucket location by using its timestamp
+    // We can determine the release's bucket location by using its timestamp
     // relative to the series starting timestamp.
     const releaseTs = new Date(release.date).getTime();
     const bucketIndex = Math.floor((releaseTs - minTime) / interval);
     const currentBucket = buckets[bucketIndex];
     const bucketEndTs = currentBucket?.end;
 
-    // Note we need to check that the release ts is within the ending bounds, solely for
-    // the last bucket, as the last buckets width can be less than the
-    // others. (i.e. if the total time difference is not perfectly divible by
-    // `desiredBuckets`, the last bucket will be the remainder)
-    if (bucketEndTs && releaseTs <= bucketEndTs) {
+    // Note that `maxTime` represents the position on the xaxis (time) of the
+    // last data point. That timestamp corresponds to data at "timestamp" +
+    // "interval", (e.g. if tooltip says 1:00 and interval between points is
+    // 30 minutes, the data collected is between 1:00 and 1:30).
+    //
+    // The release buckets are different because it has a start and end range
+    // and for its last bucket, it's ending boundary is `maxTime` to
+    // correspond to the chart. We cannot change `maxTime` for the buckets
+    // otherwise we will draw bubbles past the chart data and it would look
+    // broken.
```

Comment:
Okay I have a question about this. How broken would it look? You're doing a _lot_ of work to jam the latest releases in the last bucket (which is completely fine) but I'm curious what happens if you just don't. If you leave things as-is, and let the final bubble extend the X axis to the current timestamp, what happens? Does it add some blank space on the right side of the chart? Or something worse?

Reply:
I'll play around with it a bit more and see how bad it lokos

Reply:
🤔 yes that is gross. I'm more mad at extending past the axis than at extra space after the last segment of the line, so I feel like maybe setting a manual X axis limit would be okay here, to make them match! Anyway, up to you, I was just thinking out loud.